edxval/api.py

```diff
@@ -489,34 +490,53 @@ def copy_course_videos(source_course_id, destination_course_id):
     if source_course_id == destination_course_id:
         return
 
-    videos = Video.objects.filter(courses__course_id=unicode(source_course_id))
+    course_videos = CourseVideo.objects.select_related('video', 'video_image').filter(
+        course_id=unicode(source_course_id)
+    )
 
-    for video in videos:
-        CourseVideo.objects.get_or_create(
-            video=video,
+    for course_video in course_videos:
+        dest_course_video, __ = CourseVideo.objects.get_or_create(
+            video=course_video.video,
             course_id=destination_course_id
         )
+        try:
+            VideoImage.create_or_update(
```

Comment:
Do we need to do this in an atomic transaction covering creation if `dest_course_video` too? what are our directions if `VideoImage.create_or_update` for some reason by that time `dest_course_video` would've already been created. Do we need rollback in that case?